test/Integration/Porter/Connector/CachingConnectorTest.php

```diff
@@ -68,6 +80,59 @@ public function testCacheUsedForDifferentOptionsInstance()
         self::assertSame('foo', $this->connector->fetch('baz', clone $this->options));
     }
 
+    public function testCacheUsedForCacheKeyGenerator()
+    {
+        $this->connector->setCacheKeyGenerator(
+            \Mockery::mock(CacheKeyGenerator::class)
+                ->shouldReceive('generateCacheKey')
+                ->with('quux', $this->options->copy())
+                ->andReturn('quuz', 'quuz', 'corge')
+                ->getMock()
+        );
+
+        self::assertSame('foo', $this->connector->fetch('quux', $this->options));
+        self::assertSame('foo', $this->connector->fetch('quux', $this->options));
+        self::assertSame('bar', $this->connector->fetch('quux', $this->options));
+    }
+
+    public function testFetchThrowsInvalidCacheKeyExceptionOnNonStringCackeKey()
+    {
+        $this->setExpectedException(InvalidCacheKeyException::class, 'Cache key must be of type string.');
+
+        $this->connector->setCacheKeyGenerator(
+            \Mockery::mock(CacheKeyGenerator::class)
+                ->shouldReceive('generateCacheKey')
+                ->with('quux', $this->options->copy())
+                ->andReturn([])
+                ->getMock()
+        );
+
+        $this->connector->fetch('quux', $this->options);
+    }
+
+    public function testFetchThrowsInvalidCacheKeyExceptionOnNonPSR6CompliantCacheKey()
+    {
+        $cacheKey =  CachingConnector::RESERVED_CHARACTERS;
```

Comment:
I'm not sure why StyleCI didn't pick up the double space here but this should be disallowed. Will try to update rules later.